### PR TITLE
fix: avoid possible unhandled promise rejections

### PR DESCRIPTION
**Pull request by adrobuta**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?

Fixes the possible unhandled promise rejection edge case and adds some additional input validation.

### CODE REVIEW

File: lib/analyzer/applications/node-modules-utils.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ async function createTempProjectDir(
 const manifestName: string = "package.json";
 
 async function fileExists(path: string): Promise<boolean> {
+  if (!path) {
```
Comment (SteveShani) [inline lib/analyzer/applications/node-modules-utils.ts:35]: what is this checking?

Comment (SteveShani) [reply lib/analyzer/applications/node-modules-utils.ts:35]: Oh, if the string is "".

Comment (SteveShani) [reply lib/analyzer/applications/node-modules-utils.ts:35]: how come line 39-41 don't handle this edge case?

Comment (adrobuta) [reply lib/analyzer/applications/node-modules-utils.ts:35]: it does, removing the unnecessary check. thanks!

**Review by SteveShani — APPROVED** [review]

**Pull request merged**